syntax = "proto2";

message CSLogin
{
}

message SCLogin
{
	optional uint32 UUID = 1;
}

message CSReady
{
}

message SCReady
{
}

message SCFrameInit
{
	optional uint64 CurFrameID = 1;
	optional uint64 NxtFrameID = 2;
}

message UserFrame
{
	optional uint32 UUID = 1;
	repeated uint32 KeyInfo = 2;
}

message CSFrame
{
	optional uint32 UUID = 1;
	optional uint64 FrameID = 2;
	repeated uint32 KeyInfo = 3;
}

message SCFrame
{
	optional uint64 FrameID = 1;
	optional uint64 NextFrameID = 2;
	repeated UserFrame Users = 3;
}
